spec/std/file_spec.cr

```diff
@@ -27,11 +27,35 @@ private def it_raises_on_null_byte(operation, &block)
   end
 end
 
+private def normalize_permissions(permissions, *, directory)
+  {% if flag?(:win32) %}
+    normalized_permissions = 0o444
+    normalized_permissions |= 0o222 if permissions.bits_set?(0o200)
+    normalized_permissions |= 0o111 if directory
+    File::Permissions.new(normalized_permissions)
+  {% else %}
+    File::Permissions.new(permissions)
+  {% end %}
+end
+
+private def to_windows_path(path)
+  {% if flag?(:win32) %}
+    if path.starts_with? '/'
+      File.real_path(path)
+    else
+      path.gsub('/', '\\')
+    end
+  {% else %}
+    path
+  {% end %}
+end
+
 describe "File" do
   it "gets path" do
     path = datapath("test_file.txt")
-    file = File.new path
-    file.path.should eq(path)
+    File.open(path) do |file|
```

Comment:
All these changes from `File.new` to `File.open` are required because files are locked by default on windows, so file descriptor leaks are far more serious (and they're mostly required to make specs pass)